### PR TITLE
feat(create-mud): infer recs components from config

### PR DESCRIPTION
**Pull request by holic**
- creates components from config
- infers TS types for components from config
- removes all related codegen

### CODE REVIEW

File: packages/std-client/tsconfig.json [hunk 1]
```diff
@@ -14,5 +14,5 @@
     "skipLibCheck": true,
     "jsx": "react-jsx"
   },
-  "include": ["src"]
+  "include": ["src", "../store-sync/src/recs/common.test-d.ts"]
```
Comment (alvrs) [inline packages/std-client/tsconfig.json:17]: for context, why is this needed?

File: packages/store-sync/src/recs/common.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+describe("ConfigToRecsComponents", () => {
+  expectTypeOf<ConfigToRecsComponents<typeof storeConfig>["StoreMetadata"]>().toEqualTypeOf<
+    Component<
+      {
+        tableName: RecsType.String;
+        abiEncodedFieldNames: RecsType.String;
+      },
+      {
+        componentName: "StoreMetadata";
+        // TODO: fix config namespace so it comes back as a const
```
Comment (alvrs) [inline packages/store-sync/src/recs/common.test-d.ts:15]: should keep this in mind for #994

File: packages/store/ts/common.ts [hunk 3]
```diff
@@ -14,6 +14,12 @@ export type TableRecord<TKeySchema extends KeySchema = KeySchema, TValueSchema e
   value: SchemaToPrimitives<TValueSchema>;
 };
 
+export type ConfigFieldTypeToSchemaAbiType<T extends FieldData<string>> = T extends SchemaAbiType
+  ? T
+  : T extends `${string}[${string}]`
+  ? "uint8[]"
+  : "uint";
```
Comment (alvrs) [inline packages/store/ts/common.ts:21]: is this to turn enums into schema types? should line 21 be `uint8`?
```suggestion
export type ConfigFieldTypeToSchemaAbiType<T extends FieldData<string>> = T extends SchemaAbiType
  ? T
  : T extends `${string}[${string}]`
  ? "uint8[]"
  : "uint8";
```

**Review by alvrs — DISMISSED** [review]

File: packages/std-client/tsconfig.json (hunk 1 above)
Comment (holic) [reply packages/std-client/tsconfig.json:17]: oops, this got added by vscode after a rename

File: packages/store/ts/common.ts (hunk 3 above)
Comment (holic) [reply packages/store/ts/common.ts:21]: good catch!
